pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -139,10 +101,83 @@ func checkRollingUpdateParams(t *testing.T, deployment *appsv1.Deployment, maxUn
 	}
 }
 
-func TestDesiredRouterDeployment(t *testing.T) {
+// envData contains the name of an environment variable, whether it is expected to be present, and its expectedValue
+type envData struct {
+	name          string
+	expectPresent bool
+	expectedValue string
+}
+
+func checkDeploymentEnvironment(t *testing.T, deployment *appsv1.Deployment, expected []envData) error {
+	t.Helper()
+	var errors []error
+
+	var (
+		actualValue string
+		foundValue  bool
+	)
+
+	for _, test := range expected {
+		foundValue = false
+		for _, envVar := range deployment.Spec.Template.Spec.Containers[0].Env {
+			if envVar.Name == test.name {
+				foundValue = true
+				actualValue = envVar.Value
+				break
+			}
+		}
+		switch {
+		case !test.expectPresent && foundValue:
+			errors = append(errors, fmt.Errorf("router Deployment has unexpected %s setting: %q", test.name, actualValue))
+		case test.expectPresent && !foundValue:
+			errors = append(errors, fmt.Errorf("router Deployment is missing %v", test.name))
+		case test.expectPresent && test.expectedValue != actualValue:
+			errors = append(errors, fmt.Errorf("router Deployment has unexpected %s setting: expected %q, got %q", test.name, test.expectedValue, actualValue))
+		}
+	}
+	return utilerrors.NewAggregate(errors)
+}
+
+func TestTuningOptions(t *testing.T) {
+	ic, ingressConfig, _, apiConfig, networkConfig, _ := getRouterDeploymentComponents(t)
+
+	// Set up tuning options
+	ic.Spec.TuningOptions.ClientTimeout = &metav1.Duration{45 * time.Second}
+	ic.Spec.TuningOptions.ClientFinTimeout = &metav1.Duration{3 * time.Second}
+	ic.Spec.TuningOptions.ServerTimeout = &metav1.Duration{60 * time.Second}
+	ic.Spec.TuningOptions.ServerFinTimeout = &metav1.Duration{4 * time.Second}
+	ic.Spec.TuningOptions.TunnelTimeout = &metav1.Duration{30 * time.Minute}
+	ic.Spec.TuningOptions.TLSInspectDelay = &metav1.Duration{5 * time.Second}
+	ic.Spec.TuningOptions.HealthCheckInterval = &metav1.Duration{15 * time.Second}
+
+	deployment, err := desiredRouterDeployment(ic, ingressControllerImage, ingressConfig, apiConfig, networkConfig, false, false, nil, false)
+	if err != nil {
+		t.Fatalf("invalid router Deployment: %v", err)
+	}
+
+	// Verify tuning options
+	tests := []envData{
+		{"ROUTER_DEFAULT_CLIENT_TIMEOUT", true, "45s"},
+		{"ROUTER_CLIENT_FIN_TIMEOUT", true, "3s"},
+		{"ROUTER_DEFAULT_SERVER_TIMEOUT", true, "1m"},
+		{"ROUTER_DEFAULT_SERVER_FIN_TIMEOUT", true, "4s"},
+		{"ROUTER_DEFAULT_TUNNEL_TIMEOUT", true, "30m"},
+		{"ROUTER_INSPECT_DELAY", true, "5s"},
+		{RouterBackendCheckInterval, true, "15s"},
+	}
+
+	if err := checkDeploymentEnvironment(t, deployment, tests); err != nil {
+		t.Errorf("%s", err)
```

Comment:
```suggestion
		t.Error(err)
```
Here and below.

Reply:
👍🏼 This shows up in the goland editor as `t.Errorf("#{err}")` so I never really noticed it

Reply:
Oh, interesting!  I think I've seen that `#{err}` notation sneak into code in the past, and now I understand its provenance!